.github/workflows/check-publish.yml

```diff
@@ -0,0 +1,28 @@
+name: Check publish
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  check-publish:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+    runs-on: ${{ matrix.os }}
+    container:
+      image: paritytech/ci-unified:bullseye-1.70.0-2023-05-23-v20230706
```

Comment:
```suggestion
```

Also I'd remove container section, unless you need this specific Rust version. If no, than `github-runner` has everything needed to run next steps out of the box. (This will save couple more minutes).

All other LGTM.

Reply:
Also copied from the format check. But if it's fine to not do this I'll remove it.